test/main_test.js

```diff
@@ -0,0 +1,161 @@
+const expect = require('chai').expect;
+const WebSocket = require('ws');
+const server = require('../app.js')
+const client_util = require('../util/client_util').client_util
+let roomId = ''
+describe('room', () => {
+
+  it('Test create room', (done) => {
+    var ws = new WebSocket('ws://localhost:3000');
+    ws.onopen = function () {
+      console.log('client 1 is open')
+    }
+    ws.onmessage = function (e) {
+      jsonObj = JSON.parse(e.data);
+      switch (jsonObj.type) {
+        case 'register':
+          expect(jsonObj.playerId).to.be.string
+          var playerKey = server.edgeMultiplay.lobby.nameMap[jsonObj.playerId]
+          expect(playerKey).not.undefined
+          var playerConnection = server.edgeMultiplay.lobby.connectedClients[playerKey]
+          expect(playerConnection.address).equal(ws.address)
+          expect(playerConnection.port).equal(ws.port)
+
+          // create room
+          createReq = client_util.createRoomRequest('Player1', jsonObj.playerId, 0, 2);
+          ws.send(JSON.stringify(createReq));
+          break;
+        case 'notification':
+          expect(jsonObj.notificationText).equal('new-room-created-in-lobby')
+          break
+
+        case 'roomCreated':
+          roomId = jsonObj.room.roomId
+          expect(roomId).to.be.string
+          expect(server.edgeMultiplay.lobby.rooms.size).equal(1)
+          expect(server.edgeMultiplay.lobby.availableRooms.size).equal(1)
+          expect(server.edgeMultiplay.lobby.fullRooms.size).equal(0)
```

Comment:
it looks like some of these checks are repeated (lines 72-74 and other places). may want a helper function

Reply:
Yeah those are redundant, Thanks for catching that.